src/WebJobs.Script/Models/ExtensionPackageReference.cs

```diff
@@ -1,16 +1,17 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using System;
-using System.Collections.Generic;
-using System.Text;
-
 namespace Microsoft.Azure.WebJobs.Script.Models
 {
     /// <summary>
     /// Represents a binding extension package reference.
     /// </summary>
-    public class ExtensionPackageReference
+    public class ExtensionPackageReference : ExtensionPackageRetrieve
```

Comment:
With the introduction of flags, there immediately is a difference between what is posted/put vs what is retrieved (which was not the case earlier as Id and Version is what is posted and persisted and the same are what are retrieved too). The idea around having it as two separate classes is to reflect that HttpGet doesn't reflect what you Put/Post anymore. "ExtensionPackageRetrieve" is meant to imply that it's a contract used for "HttpGet" (I'm open to suggestions about a better name). Having it all under a single class with nullable values doesn't really make it clear about the difference in contracts. Besides,with the nullable value approach, one must ensure that the values aren't set to a non-null value, by accident in code. By using separate classes, this risk is significantly reduced (more flags can just go to ExtensionPackageReference and what HttpGet returns can remain unchanged) .. Besides, I didn't really like the idea of a primitive type like bool having a third nullable option but that's probably debatable.. :) Please let me know if you have more opinions about this, I'll be happy to discuss.

---
In reply to: [299277136](https://github.com/Azure/azure-functions-host/pull/4617#discussion_r299277136) [](ancestors = 299277136)

Reply:
> The idea around having it as two separate classes is to reflect that HttpGet doesn't reflect what you Put/Post anymore. "ExtensionPackageRetrieve" is meant to imply that it's a contract used for "HttpGet" (I'm open to suggestions about a better name). Having it all under a single class with nullable values didn't really make it obvious to me at first look, that the contracts are different. 

May be i am missing something here. But i am not sure how this is achieved since you are using the same contract for GET and PUT. I do agree that having a nullable boolean is not a clean or an ideal approach.

I think we do need to have separate contracts for `GET` vs `PUT/POST`. Mainly because the `DeferAppActivationToServer` has no context or meaning in the `GET` call. Also we do not store it anywhere. You would always get `false` even when it is not set or set to true during the `post/put` call. Also i think we should have the same terminology to set offline state / set online state. This way it would be easier to understand what is causing the change in host state and also would maintain the same terminology for clients as well. We could use a string type for the flag that explicitly mentions the post install state of host (`offline`/`online`/ `do nothing`) could be used here. This would allow the client to have the option of `do nothing` which would map to the default value of string which is `null`.

Reply:
Modified this to use flags as discussed